src/openrct2/EditorObjectSelectionSession.cpp

```diff
@@ -375,6 +376,24 @@ void reset_selected_object_count_and_size()
     }
 }
 
+void finish_object_selection()
+{
+    if (gScreenFlags & SCREEN_FLAGS_TRACK_DESIGNER)
+    {
+        set_every_ride_type_invented();
+        set_every_ride_entry_invented();
+        gS6Info.editor_step = EDITOR_STEP_ROLLERCOASTER_DESIGNER;
+        gfx_invalidate_screen();
+    }
+    else
+    {
+        set_every_ride_type_invented();
```

Comment:
This _still_ doesn't reflect what the old if-else did: https://github.com/OpenRCT2/OpenRCT2/pull/11010/files#diff-af125fc02d11204f8f2787620a406bb7L238